rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,153 @@
+import com.netflix.spinnaker.rosco.manifests.TemplateUtils;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Component
+@Slf4j
+public class KustomizeTemplateUtils extends TemplateUtils {
+
+  private HashSet<String> filesToDownload = new HashSet<String>();
+  private static final String KUSTOMIZATION_FILE = "kustomization";
+
+  @Autowired
+  private KustomizationFileReader kustomizationFileReader;
+
+
+  public BakeRecipe buildBakeRecipe(BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+    BakeRecipe result = new BakeRecipe();
+    result.setName(request.getOutputName());
+    filesToDownload = new HashSet<>();
+
+    List<Path> valuePaths = new ArrayList<>();
+    List<Artifact> inputArtifacts = request.getInputArtifacts();
+    if (inputArtifacts == null || inputArtifacts.isEmpty()) {
+      throw new IllegalArgumentException("At least one input artifact must be provided to bake");
+    }
+    List<Artifact> artifacts = new ArrayList<>();
+    Artifact artifact = inputArtifacts.get(0);
+    String templatePath = getKustomizationPath(env,artifact);
+    try {
+      Path path = Paths.get(artifact.getReference());
+        artifact.setReference(path.toString());
+      for (String s : getFilesFromGithub(artifact)) {
+        Artifact a = new Artifact();
+        a.setReference(s);
+        a.setArtifactAccount(artifact.getArtifactAccount());
+        a.setCustomKind(artifact.isCustomKind());
+        a.setLocation(artifact.getLocation());
+        a.setMetadata(artifact.getMetadata());
+        a.setName(artifact.getName());
+        a.setProvenance(artifact.getProvenance());
+        a.setType(artifact.getType());
+        a.setUuid(artifact.getUuid());
+        a.setVersion(artifact.getVersion());
+        artifacts.add(a);
+      }
+    } catch (IOException e) {
+      log.error("Error setting references in artifacts from GitHub " + e.getMessage());
+    }
+    try {
+      for(Artifact ar : artifacts){
+        downloadArtifactToTmpFileStructure(env, ar);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to fetch helm template: " + e.getMessage(), e);
+    }
+
+    List<String> command = new ArrayList<>();
+    command.add("kustomize");
+    command.add("build");
+    command.add( templatePath);
+
+
+      result.setCommand(command);
+
+    return result;
+  }
+
+  private HashSet<String> getFilesFromGithub(Artifact artifact) throws IOException {
```

Comment:
Change `getFilesFromGithub` to something more generic. Maybe `getFileFromArtifact`. While Github is our base use case, Artifacts provide a more generic interface for getting files.

Reply:
changed